src/wasm/src/dispatcher.rs

```diff
@@ -0,0 +1,439 @@
+                    let communications = self.database.get_communications_at_airport(params.airport_ident)?;
+
+                    t.borrow_mut().status = TaskStatus::Success(Some(serde_json::to_value(communications)?));
+
+                    Ok(())
+                }),
+                functions::FunctionType::GetGlsNavaidsAtAirport => Dispatcher::execute_task(task.clone(), |t| {
+                    let params = t.borrow().parse_data_as::<params::GetAtAirportParams>()?;
+                    let navaids = self.database.get_gls_navaids_at_airport(params.airport_ident)?;
+
+                    t.borrow_mut().status = TaskStatus::Success(Some(serde_json::to_value(navaids)?));
+
+                    Ok(())
+                }),
+                functions::FunctionType::GetPathPointsAtAirport => Dispatcher::execute_task(task.clone(), |t| {
+                    let params = t.borrow().parse_data_as::<params::GetAtAirportParams>()?;
+                    let pathpoints = self.database.get_path_points_at_airport(params.airport_ident)?;
+
+                    t.borrow_mut().status = TaskStatus::Success(Some(serde_json::to_value(pathpoints)?));
+
+                    Ok(())
+                }),
+            }
+        }
+
+        // Process completed tasks (everything should at least be in progress at this point)
+        queue.retain(|task| {
+            if let TaskStatus::InProgress = task.borrow().status {
+                return true;
+            }
+
+            let status: functions::FunctionStatus;
+            let data: Option<serde_json::Value>;
+
+            let (status, data) = match task.borrow().status {
+                TaskStatus::Success(ref result) => {
+                    status = functions::FunctionStatus::Success;
+                    data = result.clone();
+                    (status, data)
+                },
+                TaskStatus::Failure(ref error) => {
+                    status = functions::FunctionStatus::Error;
+                    data = Some(error.clone().into());
+                    (status, data)
+                },
+                _ => unreachable!(), // This should never happen
+            };
+
+            let json = functions::FunctionResult {
+                id: task.borrow().id.clone(),
+                status,
+                data,
+            };
+
+            if let Ok(serialized_json) = serde_json::to_string(&json) {
+                CommBus::call("NAVIGRAPH_FunctionResult", &serialized_json, CommBusBroadcastFlags::All);
+            }
+            false
+        });
+    }
+
+    /// Executes a task and handles the result (sets the status of the task)
+    fn execute_task<F>(task: Rc<RefCell<Task>>, task_operation: F)
+    where
+        F: FnOnce(Rc<RefCell<Task>>) -> Result<(), Box<dyn std::error::Error>>,
+    {
+        match task_operation(task.clone()) {
+            Ok(_) => (),
+            Err(e) => {
+                println!("Task failed: {}", e);

```

Comment:
We should preface all `println!`s with something unique, such as `[NAVIGRAPH]` for easier searching in the sim console